.github/workflows/snl-ci.yml

```
@@ -0,0 +1,62 @@
+name: SNL-CI
+
+on: 
+  pull_request:
+    paths-ignore:
+    - '**/*.rst'
+    - '**/*.md'
+    - '**/requirements.txt'
+    - '**/*.py'
+    - 'docs/**'
+    types: [ opened, reopened, synchronize ]
+
+permissions:
+  contents: none
+
+# Cancels any in progress 'workflow' associated with this PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  :CUDA_12_2_CPP20
+    name: SNL_CUDA_NVCC_12_2_CPP20
+    runs-on: [snl-kk-env-cuda-12.2.0-gcc-11.3.0-latest]
+    
+    steps:
+      - name: checkout_kokkos
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          repository: kokkos/kokkos
+          ref: ${{ github.base_ref }}
+          path: kokkos
+
+      - name: configure_kokkos
+        run: |
+          mkdir -p kokkos/{build,install}
+          nvidia-smi
+          cd kokkos/build
+          cmake -DCMAKE_CXX_COMPILER=$PWD/../bin/nvcc_wrapper \
+            -DCMAKE_INSTALL_PREFIX=$PWD/../install \
+            -DCMAKE_CXX_STANDARD=20 \
+            -DKokkos_ENABLE_CUDA=ON \
+            -DKokkos_ARCH_HOPPER90=ON \
+            -DCMAKE_VERBOSE_MAKEFILE=ON \
+            -DCMAKE_CXX_EXTENSIONS=OFF \
+            -DBUILD_SHARED_LIBS=OFF \
+            -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
+            -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF \
+            -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
+            -DKokkos_ENABLE_TESTS=ON \
+            -DKokkos_ENABLE_EXAMPLES=ON \
+            -DKokkos_ENABLE_BENCHMARKS=ON \
+            ..
+
+      - name: build_and_install_kokkos
+        working-directory: kokkos/build
+        run: make -j36 install
+
+      - name: test_kokkos
+        working-directory: kokkos/build
+        run: ctest --output-on-failure -V --timeout 3600
```

`--output-on-failure` is redundant if  `-V` is used
```suggestion
        run: ctest --output-on-failure --timeout 3600
```